packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -0,0 +1,373 @@
+  /// is also null, the value of [textStyle] will be used. If this is also null,
+  /// defaults to the `bodyLarge` text style from the current [Theme].
+  /// The default text color is [ColorScheme.onSurfaceVariant].
+  final MaterialStateProperty<TextStyle?>? hintStyle;
+
+  @override
+  State<SearchBar> createState() => _SearchBarState();
+}
+
+class _SearchBarState extends State<SearchBar> {
+  late final MaterialStatesController _internalStatesController;
+  late final FocusNode _focusNode;
+
+  @override
+  void initState() {
+    super.initState();
+    _internalStatesController = MaterialStatesController();
+    _internalStatesController.addListener(() {
+      setState(() {});
+    });
+    _focusNode = widget.focusNode ?? FocusNode();
+  }
+
+  @override
+  void dispose() {
+    _internalStatesController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final TextDirection textDirection = Directionality.of(context);
+    final ColorScheme colorScheme = Theme.of(context).colorScheme;
+    final IconThemeData iconTheme = IconTheme.of(context);
+    final SearchBarThemeData searchBarTheme = SearchBarTheme.of(context);
+    final SearchBarThemeData defaults = _SearchBarDefaultsM3(context);
+
+    T? resolve<T>(MaterialStateProperty<T>? property) {
+      return property?.resolve(_internalStatesController.value);
+    }
+
+    final TextStyle? effectiveTextStyle = resolve<TextStyle?>(widget.textStyle ?? searchBarTheme.textStyle ?? defaults.textStyle);
+    final TextStyle? effectiveHintStyle = resolve<TextStyle?>(widget.hintStyle ?? searchBarTheme.hintStyle ?? widget.textStyle ?? searchBarTheme.textStyle ?? defaults.hintStyle);
+    final double? effectiveElevation = resolve<double?>(widget.elevation ?? searchBarTheme.elevation ?? defaults.elevation);
+    final Color? effectiveShadowColor = resolve<Color?>(widget.shadowColor ?? searchBarTheme.shadowColor ?? defaults.shadowColor);
+    final Color? effectiveBackgroundColor = resolve<Color?>(widget.backgroundColor ?? searchBarTheme.backgroundColor ?? defaults.backgroundColor);
+    final Color? effectiveSurfaceTintColor = resolve<Color?>(widget.surfaceTintColor ?? searchBarTheme.surfaceTintColor ?? defaults.surfaceTintColor);
+    final OutlinedBorder? effectiveShape = resolve<OutlinedBorder?>(widget.shape ?? searchBarTheme.shape ?? defaults.shape);
+    final BorderSide? effectiveSide = resolve<BorderSide?>(widget.side ?? searchBarTheme.side ?? defaults.side);
+    final EdgeInsetsGeometry? effectivePadding = resolve<EdgeInsetsGeometry?>(widget.padding ?? searchBarTheme.padding ?? defaults.padding);
+    final MaterialStateProperty<Color?>? effectiveOverlayColor = widget.overlayColor ?? searchBarTheme.overlayColor ?? defaults.overlayColor;
+
+    final bool isDark = Theme.of(context).brightness == Brightness.dark;
+    bool isIconThemeColorDefault(Color? color) {
+      if (isDark) {
+        return color == kDefaultIconLightColor;
```

Comment:
Shouldn't kDefaultIconLightColor etc come from the theme's ColorScheme?

Reply:
Nevermind that ^^^ comment, I understand this now. Someday we should provide an IconTheme utility that does this.